src/uploader.py

```diff
@@ -0,0 +1,108 @@
+"""Collections of different upload methods."""
+
+import logging
+import socket
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, List, Optional
+
+import paramiko
+from paramiko.sftp_client import SFTPClient
+
+logger = logging.getLogger(__name__)
+
+
+class UnknownUploaderError(Exception):
+    """Unknown uploader exception."""
+
+
+class Uploader:
```

Comment:
**suggestion:** 

I will suggest this class should be simple like a function.
Because here we mix the factory and strategy pattern together.


```python
def get_uploader(strategy: str, **kwargs) -> BaseUploader:
   if strategy == "sftp":
      return SFTPUploader(**kwargs)
   elif strategy == "scp":
     ...
```

Separate simple factory function and Uploader will give us more flexible in the future. Also more easy to write tests.